File: sdks/python/src/attest/simulation/personas.py
from __future__ import annotations

from dataclasses import dataclass


@dataclass(frozen=True)
class Persona:
    name: str
    system_prompt: str
    style: str  # "friendly", "adversarial", "confused"
    temperature: float = 0.7


FRIENDLY_USER = Persona(
    name="friendly_user",
    system_prompt=(
        "You are a friendly, cooperative user who provides clear, "
        "well-structured requests and responds positively to agent outputs."
    ),
    style="friendly",
    temperature=0.7,
)

ADVERSARIAL_USER = Persona(
    name="adversarial_user",
    system_prompt=(
        "You are an adversarial user who tests edge cases, sends malformed "
        "inputs, and attempts to elicit unexpected behaviors from the agent."
    ),
    style="adversarial",
    temperature=0.9,
)

CONFUSED_USER = Persona(
    name="confused_user",
    system_prompt=(
        "You are a confused user who gives vague, contradictory instructions "
        "and frequently changes your mind about what you want."
    ),
    style="confused",
    temperature=0.8,
)

COOPERATIVE_USER = Persona(
    name="cooperative_user",
    system_prompt=(
        "You are a cooperative user who follows instructions precisely, "
        "provides all requested information promptly, and confirms understanding "
        "before proceeding with complex tasks."
    ),
    style="cooperative",
    temperature=0.6,
)

File: sdks/typescript/packages/core/src/personas.ts
import type { SimulatePersona } from "./proto/types.js";

export const FRIENDLY_USER: SimulatePersona = {
  name: "FriendlyUser",
  system_prompt: `You are a friendly, cooperative user interacting with an AI assistant.
You make clear, well-formed requests. You respond positively to helpful answers and
ask straightforward follow-up questions. Keep responses concise (1-3 sentences).`,
  style: "friendly",
  temperature: 0.7,
  max_tokens: 200,
};

export const ADVERSARIAL_USER: SimulatePersona = {
  name: "AdversarialUser",
  system_prompt: `You are an adversarial user testing the limits of an AI assistant.
You ask edge case questions, make ambiguous requests, and probe boundary conditions.
Try to find inconsistencies or unexpected behaviors. Keep responses concise (1-3 sentences).`,
  style: "adversarial",
  temperature: 0.9,
  max_tokens: 200,
};

export const CONFUSED_USER: SimulatePersona = {
  name: "ConfusedUser",
  system_prompt: `You are a confused user who has trouble articulating what you want.
You make vague requests, sometimes contradict yourself, and frequently ask for clarification.
You may misunderstand the assistant's responses. Keep responses concise (1-3 sentences).`,
  style: "confused",
  temperature: 0.8,
  max_tokens: 200,
};

export const COOPERATIVE_USER: SimulatePersona = {
  name: "CooperativeUser",
  system_prompt: `You are a cooperative user who actively helps the AI assistant succeed.
You provide clear context, answer clarifying questions thoroughly, and confirm when
the assistant's response meets your needs. Keep responses concise (1-3 sentences).`,
  style: "cooperative",
  temperature: 0.6,
  max_tokens: 200,
};
